src/js_sync.hpp

```diff
@@ -303,7 +303,7 @@ void SyncClass<T>::populate_sync_config(ContextType ctx, ObjectType realm_constr
 
         // FIXME - use make_shared
         config.sync_config = std::shared_ptr<SyncConfig>(
-            new SyncConfig(shared_user, raw_realm_url, SyncSessionStopPolicy::AfterChangesUploaded, handler)
+                                                         new SyncConfig({shared_user, raw_realm_url, SyncSessionStopPolicy::AfterChangesUploaded, handler, [=](int error_code, std::string message, SyncSessionError) {}})
```

Comment:
The extra parens and empty lambda seem unnecessary at first glance. Does `new SyncConfig{shared_user, raw_realm_url, SyncSessionStopPolicy::AfterChangesUploaded, handler}` not work?